jazz-ui/src/app/pages/environment-detail/environment-detail.component.ts

```diff
@@ -198,59 +198,51 @@ api_doc_name:string='';
 
   };
 
-    testApi(type){
-        switch(type){
-            case 'api': 
-            if(environment.envName == "oss"){
-              var SwaggerUrl="http://editor.swagger.io/?url="+this.api_doc_name+"/"+this.service.domain +"/"+ this.service.name +"/"+this.envSelected+"/swagger.json"
-              window.open(SwaggerUrl);
-            } 
-            else{
-              window.open('/test-api?service=' + this.service.name + '&domain='+ this.service.domain + '&env=' +this.envSelected);
-
-            }        
-            
-            break;
-
-            case 'website' :
-            if(this.endpoint_env!=(undefined||'')){
-              window.open(this.endpoint_env);    
-            }
-            break;
-            case 'function' :
-            if(this.endpoint_env!=(undefined||'')){
-              window.open('/404');    
-            }
-            break;
-            case 'lambda' :
-            if(this.endpoint_env!=(undefined||'')){
-              window.open('/404');    
-            }
-            break;
+  testApi(type) {
+    switch (type) {
+      case 'api':
+        if (environment.envName == "oss") {
+          var apiPath =  this.api_doc_name + "/" + this.service.domain + "/" + this.service.name + "/" + this.envSelected + "/swagger.json";
+          var SwaggerUrl = "http://editor.swagger.io/?url=" + apiPath;
+          window.open(SwaggerUrl);
+        }
+        else {
+          window.open('/test-api?service=' + this.service.name + '&domain=' + this.service.domain + '&env=' + this.envSelected);
+        }
+
+        break;
+
+      case 'website' :
+        if (this.endpoint_env != (undefined || '')) {
+          window.open(this.endpoint_env);
         }
+        break;
+      case 'function' :
```

Comment:
case function:
case lambda:
      this.setSidebar('try-service');
     break;